geomesa-fs/geomesa-fs-tools/src/test/scala/org/locationtech/geomesa/fs/tools/ingest/CompactCommandTest.scala

```diff
@@ -12,143 +12,182 @@ import java.nio.file.{Files, Path}
 
   import org.locationtech.geomesa.fs.storage.common.RichSimpleFeatureType
 
   sequential
 
-  var cluster: MiniDFSCluster = _
-  var directory: String = _
-  val typeName = "orc"
-
   val tempDir: Path = Files.createTempDirectory("compactCommand")
 
-  val pt: Point              = WKTUtils.read("POINT(0 0)").asInstanceOf[Point]
-  val line: LineString       = WKTUtils.read("LINESTRING(0 0, 1 1, 4 4)").asInstanceOf[LineString]
-  val polygon: Polygon       = WKTUtils.read("POLYGON((10 10, 10 20, 20 20, 20 10, 10 10), (11 11, 19 11, 19 19, 11 19, 11 11))").asInstanceOf[Polygon]
-  val mpt: MultiPoint        = WKTUtils.read("MULTIPOINT((0 0), (1 1))").asInstanceOf[MultiPoint]
-  val mline: MultiLineString = WKTUtils.read("MULTILINESTRING ((0 0, 1 1), \n  (2 2, 3 3))").asInstanceOf[MultiLineString]
-  val mpolygon: MultiPolygon = WKTUtils.read("MULTIPOLYGON(((0 0, 1 0, 1 1, 0 0)), ((10 10, 10 20, 20 20, 20 10, 10 10), (11 11, 19 11, 19 19, 11 19, 11 11)))").asInstanceOf[MultiPolygon]
+  val encodings = Seq("parquet", "orc")
 
-  val sft: SimpleFeatureType = SimpleFeatureTypes.createType(typeName, "name:String,age:Int,dtg:Date," +
-    "*geom:MultiLineString:srid=4326,pt:Point,line:LineString,poly:Polygon,mpt:MultiPoint,mline:MultiLineString,mpoly:MultiPolygon")
-  sft.setScheme("daily")
+  var cluster: MiniDFSCluster = _
+  var directory: String = _
 
-  val features: Seq[ScalaSimpleFeature] = Seq.tabulate(10) { i =>
-    ScalaSimpleFeature.create(sft, s"$i", s"test$i", 100 + i, s"2017-06-0${5 + (i % 3)}T04:03:02.0001Z", s"MULTILINESTRING((0 0, 10 10.$i))",
-      pt, line, polygon, mpt, mline, mpolygon)
+  val pt       = WKTUtils.read("POINT(0 0)")
+  val line     = WKTUtils.read("LINESTRING(0 0, 1 1, 4 4)")
+  val polygon  = WKTUtils.read("POLYGON((10 10, 10 20, 20 20, 20 10, 10 10), (11 11, 19 11, 19 19, 11 19, 11 11))")
+  val mpt      = WKTUtils.read("MULTIPOINT((0 0), (1 1))")
+  val mline    = WKTUtils.read("MULTILINESTRING ((0 0, 1 1), (2 2, 3 3))")
+  val mpolygon = WKTUtils.read("MULTIPOLYGON(((0 0, 1 0, 1 1, 0 0)), ((10 10, 10 20, 20 20, 20 10, 10 10), (11 11, 19 11, 19 19, 11 19, 11 11)))")
+
+  val sfts = encodings.map { name =>
+    val sft = SimpleFeatureTypes.createType(name,
+      "name:String,age:Int,dtg:Date," +
+        "*geom:MultiLineString:srid=4326,pt:Point,line:LineString," +
+          "poly:Polygon,mpt:MultiPoint,mline:MultiLineString,mpoly:MultiPolygon")
+    sft.setEncoding(name)
+    sft.setScheme("daily")
+    sft
   }
 
-  val features2: Seq[ScalaSimpleFeature] = Seq.tabulate(10) { j =>
-    val i = j+10
-    ScalaSimpleFeature.create(sft, s"$i", s"test$i", 100 + i, s"2017-06-0${5 + (j % 3)}T04:03:02.0001Z", s"MULTILINESTRING((0 0, 10 10.$j))",
-      pt, line, polygon, mpt, mline, mpolygon)
-  }
+  val numFeatures = 10000
+  val targetFileSize = 12000L // kind of a magic number, in that it divides up the features into files fairly evenly with no remainder
 
-  def getDataStore: DataStore = {
+  lazy val ds = {
```

Comment:
Can the one FSDS read from both Orc and Parquet formats?

Reply:
yes!

Reply:
Neat!